contexts/SocketContext.tsx

```diff
@@ -0,0 +1,34 @@
+import { createContext, ReactNode, useEffect, useState } from "react";
+import { io, Socket } from "socket.io-client";
+
+type SocketContextType = {
+  socket: Socket;
+};
+
+export const SocketContext = createContext<SocketContextType>(null);
```

Comment:
Won't this produce a type script error as the context type would be `SocketContextType | null`?

Reply:
Nope, TypeScript is fine with this.
Maybe because `null` is just the default value and will later be socket?

Reply:
You are right in that it is effectively of type `SocketContextType` when you use it, but assigning `null` to something of type `SocketContextType` is a type contract violation, nonetheless. So, how does the compiler know? 🤔 will have to do some more digging.

Reply:
Seems that you can use `null` everywhere another type is expected. This is probably configurable in tsconfig, didn't know about that, before